File: testpilot-api/app/schemas/project.py
from pydantic import BaseModel


class ProjectCreate(BaseModel):
    name: str
    system_type: str
    test_types: list[str]
    script_language: str
    output_language: str = "中文"


class ProjectResponse(ProjectCreate):
    id: str
    created_at: str
    updated_at: str
    last_generated_module: str | None = None

File: testpilot-web/src/types/project.ts
export type SystemType = "Web" | "App" | "API" | "小程序" | "后台系统";
export type ScriptLanguage = "Python" | "Java";

export interface Project {
  id: string;
  name: string;
  system_type: SystemType | string;
  test_types: string[];
  script_language: ScriptLanguage | string;
  output_language: string;
  created_at?: string;
  updated_at?: string;
  last_generated_module?: string | null;
}
